### ISSUE TITLE
default support for ptex ?

### ISSUE THREAD

**Issue by davidcarlisle** [latex3/graphics-cfg#1]
With the graphics.cfg as maintained here you get an error from:

```
\documentclass{article}

\usepackage{graphicx}
%\usepackage[dvipdfmx]{graphicx}

\begin{document}

abc

\includegraphics{example-image.pdf}

\end{document}
```

Should this graphics.cfg detect ptex and variants and default to dvipdfmx ? (@aminophen )

**Comment by aminophen** [comment]
I’m afraid, changing the default at this point of time will break many existing documents. Many casual users of pTeX/upTeX are using dvipdfmx as a driver, but unignorable number of users (especially those working in publishing companies) are using dvips. Also, we don't want the default difference between (u)platex and latex in DVI mode.

----

BTW, recently I observed some issues in Japanese community about driver selection.

[1] Recent expl3 has driver options, whose interface is not consistent with 2e convention. Many Japanese users of pTeX/upTeX+dvipdfmx choose to write

\documentclass[dvipdfmx]{some-class}% global option

instead of individual package options

\usepackage[dvipdfmx]{graphicx}
\usepackage[dvipdfmx]{hyperref}

for simplicity. OTOH, expl3 does not reflect the global option because its rather new syntax requires [driver=dvipdfmx]. I hope the global option is passed to expl3 correctly.

[2] When a user chooses individual package options rather than global,

\usepackage[dvipdfmx]{graphicx}
\usepackage{some driver-dependent package}% => contains \RequirePackage{color}

This code loads both dvipdfmx.def (from graphicx) and dvips.def (from color), which is definitely undesirable. This double loading will never give a correct result, so we hope some error or warning about that.

----

I'm not sure where to post the above issues. Any advice?

**Comment by josephwright** [comment]
@aminophen  My impression was that upTeX really only 'made sense' with `dvipdfmx` due to font support: is this not correct?

On the option business, we are working on some changes for the January 2020 release of LaTeX2e and `expl3`: I think it is likely we will be able to address those issues then.

**Comment by aminophen** [comment]
No; by default TeX Live uses [IPAex](https://www.ctan.org/pkg/ipaex) for Japanese, and those TrueType fonts can be correctly embedded using (u)pTeX-dvips-ps2pdf. Note that you will need setting up Ghostscript by running the script [cjk-gs-integrate](https://ctan.org/pkg/cjk-gs-integrate) before running ps2pdf.

**Comment by davidcarlisle** [comment]
@aminophen  yes I wondered if that would be the case, which is why I pinged you before changing anything, so I guess we won't change this (let us know if there is anything we could do to make this better for ptex)

The fact that color and graphics can separately specify a back end but they share back end .def files so if you can end up with inconsistent loading is a general failing in the design of those packages (sorry:-)  Again it is compatibility issues that make be wary of changing anything, but I think probably they should detect that a def file is already loaded and so not load a second one, and warn about inconsistent options.  Probably not for the release planned soon, but we should look at that for the following release.

**Comment by josephwright** [comment]
@aminophen Useful information: I guess we (I) perhaps should alter the standard setting for the `expl3` drivers in that case, as part of unifying the interfaces. (Unlike `color`/`graphics`, of course,  `expl3` has a single-point-of-truth for backend selection.)
